### DISCUSSION TITLE
Time for a new nlmod version (0.7.0)?

### DISCUSSION THREAD

**Discussion by OnnoEbbens** [gwmod/nlmod#267] [General]
Many improvements have been made in the dev branch since the last version 0.6.1 was released on June 26th. I think it is time for a new nlmod version. What do you think @bdestombe, @dbrakenhoff, @rubencalje?

**Comment by bdestombe** [comment]
Great! But there have been a few breaking changes. Is a minor bump of the version number enough?

**Reply by OnnoEbbens** [reply]
I would prefer to use minor versions with breaking changes until we reach a first stable 1.x.x version. I think we will have some more breaking changes coming up in the near future and I don't like to do a major version bump for each one of them. 

We could off course invest more in backward comparability so we don't have to do as much major version bumps. But I am afraid that trying to guarantee backward compatibility at this moment will take too much of our time, slow down development and may lead to sub-optimal solutions.

So I would propose to work towards a first stable 1.x.x. version and guarantee backward compatibility from that version. Until then we can try to be as backward compatible as possible but at the same time feel free to make breaking changes if they are serious improvements.

Curious what you think about this

**Reply by bdestombe** [reply]
Perfect Onno, Sounds good to me!
